pydevices/HtsDevices/acq435st.py

```diff
@@ -145,32 +157,9 @@ def init(self):
             coeff.record = coeffs[i]
             offset = self.__getattr__('input_%2.2d_offset'%(i+1))
             offset.record = offsets[i]
-        rundata = {}
         self.running.on=True
-        fd = tempfile.NamedTemporaryFile(mode='w', suffix='.py', prefix='tmp', dir='/tmp', delete=False)
-        rundata['tmp_file'] = fd.name
-        logfd = tempfile.NamedTemporaryFile(mode='w', suffix='.log', prefix='tmp', dir='/tmp', delete=False)
-        rundata['log_file'] = logfd.name
-        fd.write('from MDSplus import Tree\n')
-        fd.write('ans = Tree("%s", %d).getNode(r"%s").stream()\n' % (self.tree.name, self.tree.shot, self.path))
-        fd.write('exit()\n')
-        if self.debugging():
-            print('script name is %s' % fd.name)
-        fd.close()
-        logfd = open(rundata['log_file'], 'w')
-        logfd.write('\n---------------------------------------------------------------------\n')
-        logfd.write('\tACQ435ST\n')
-        pid = subprocess.Popen("%s %s %d" % 
-                               (sys.executable, fd.name, os.getpid()), 
-                                shell=True, 
-                                stdout=logfd, 
-                                stderr=logfd).pid
-        if self.debugging():
-            print('The streaming pid is %d' % pid)
-            print('Parent Pid is %d' % os.getpid())
-        fd.close()
-        logfd.close()
-        self.running.record = rundata
+        thread = self.Worker(self)
+        thread.start()
         return 1
```

Comment:
No need to return a status code. simply raise an exception if not ok

Reply:
I saw this comment of yours about Fernando's PR last week.  Turns out do /meth in TCL is a bit unhappy if you do not return something odd.  
This could probably be fixed there but for now, I want to leave it.